compiler/src/iree/compiler/Dialect/Flow/Transforms/AnnotateDispatches.cpp

```diff
@@ -179,6 +179,22 @@ static std::string summarizeLinalgOp(linalg::LinalgOp op) {
     }
   }
 
+  // Categorize linalg.generic ops better.
+  if (prefix.empty() && isa<linalg::GenericOp>(op)) {
+    if (llvm::all_of(op.getIndexingMapsArray(),
+                     [](AffineMap m) { return m.isIdentity(); })) {
+      prefix = "elementwise";
+    } else if (llvm::all_of(op.getIndexingMapsArray(),
+                            [](AffineMap m) { return m.isMinorIdentity(); })) {
+      // We have checked that this is not pure elementwise in the above.
+      prefix = "broadcast";
+    } else if (linalg::isaContractionOpInterface(op)) {
+      prefix = "contract";
+    } else if (linalg::isaConvolutionOpInterface(op)) {
+      prefix = "conv";
```

Comment:
how does this work with fusions? Do we just take the first name when walking the IR? Just wondering about cases where we could have a contraction + elementwise, and it would be misleading if the dispatch name said elementwise in that case.

Reply:
Whichever op has a higher "cost" as computed by a heuristic is chosen to produce the summary text. See `estimateLinalgOpCost` and `estimateLinalgExtOpCost` above (roughly "whichever op iterates over the most elements").

You can run with `--debug-only=iree-dispatch` to see which ops are evaluated and their computed costs